### DISCUSSION TITLE
React Native Firebase Emulator on LAN environment

### DISCUSSION THREAD

**Discussion by ithustle** [invertase/react-native-firebase#5750] [Q&A] [answered]
Hi there,

Is it possible run `react-native-firebase` emulator with a local network server IP?
Using `auth().useEmulator(`http://${SERVER_IP}:9099`)` the `Auth` instance doesn't communicate with Firebase Auth Emulator and using `firestore().useEmulator(SERVER_IP, 8080)` it return the following message:

```
Error: firebase.firestore().useEmulator() takes a non-empty host and port
```

**Comment by mikehardy** [comment, marked as answer]
Unfortunately the emulator config is a little disjoint - auth takes a slightly different format than the others, but you may see how we configure them here, I believe you can replace "localhost" with a server IP in any of those and it should work? 

https://github.com/invertase/react-native-firebase/blob/4c7840a86f9af7e7d2853749dc57a917fe9709fd/tests/app.js#L49-L53

**Reply by ithustle** [reply]
Hi @mikehardy. 
Thanks for reply.

I did exactly the same:

```
const SERVER_IP = 'server_ip_on_my_lan';
...
if (__DEV__) {
        firestore().settings({ ssl: false, cacheSizeBytes: firestore.CACHE_SIZE_UNLIMITED });
        functions().useFunctionsEmulator(`http://${SERVER_IP}:5001`);
        firestore().useEmulator(SERVER_IP, 8080);
        auth().useEmulator(`http://${SERVER_IP}:9099`);
        storage().useEmulator(SERVER_IP, 9199);
    } else {
        firestore().settings({ cacheSizeBytes: firestore.CACHE_SIZE_UNLIMITED });
    }
```
...and still throw this message: `Error: firebase.firestore().useEmulator() takes a non-empty host and port, js engine: hermes`.

Any tips?

**Reply by ithustle** [reply]
Replacing the variable `SERVER_IP` to IP works. :/
